applications/IgaApplication/custom_elements/shell_6p_element.h

```diff
@@ -0,0 +1,667 @@
+//                   Maram Alkhlaifat
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+#include "utilities/math_utils.h"
+
+// Application includes
+#include "iga_application_variables.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+/// Short class definition.
+/** Reissner-Mindlin shell based on degenerated solid (Benson et al.)
+*/
+class Shell6pElement
+    : public Element
+{
+protected:
+
+    /// Internal variables used for metric transformation
+    struct KinematicVariables
+    {
+        // covariant metric
+        array_1d<double, 3> a_ab_covariant;
+        array_1d<double, 3> b_ab_covariant;
+
+        //base vector 1
+        array_1d<double, 3> a1;
+        //base vector 2
+        array_1d<double, 3> a2;
+        //base vector 3 normalized
+        array_1d<double, 3> a3;
+        //not-normalized base vector 3
+        array_1d<double, 3> a3_tilde;
+
+        //differential area
+        double dA;
+
+        /**
+        * The default constructor
+        * @param Dimension: The size of working space dimension
+        */
+        KinematicVariables(std::size_t Dimension)
+        {
+            noalias(a_ab_covariant) = ZeroVector(Dimension);
+            noalias(b_ab_covariant) = ZeroVector(Dimension);
+
+            noalias(a1) = ZeroVector(Dimension);
+            noalias(a2) = ZeroVector(Dimension);
+            noalias(a3) = ZeroVector(Dimension);
+
+            noalias(a3_tilde) = ZeroVector(Dimension);
+
+            dA = 1.0;
+        }
+    };
+
+    /**
+    * Internal variables used in the constitutive equations
+    */
+    struct ConstitutiveVariables
+    {
+        Vector StrainVector;
+        Vector StressVector;
+        Matrix ConstitutiveMatrix;
+
+        /**
+        * @param StrainSize: The size of the strain vector in Voigt notation
+        */
+        ConstitutiveVariables(std::size_t StrainSize)
+        {
+            StrainVector = ZeroVector(StrainSize);
+            StressVector = ZeroVector(StrainSize);
+            ConstitutiveMatrix = ZeroMatrix(StrainSize, StrainSize);
+        }
+    };
+
+    /**
+    * Internal variables used in the constitutive equations
+    */
+    struct SecondVariations
+    {
+        Matrix B11;
+        Matrix B22;
+        Matrix B12;
+
+        /**
+        * The default constructor
+        * @param StrainSize: The size of the strain vector in Voigt notation
+        */
+        SecondVariations(const int& mat_size)
+        {
+            B11 = ZeroMatrix(mat_size, mat_size);
+            B22 = ZeroMatrix(mat_size, mat_size);
+            B12 = ZeroMatrix(mat_size, mat_size);
+        }
+    }; 
+
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Counted pointer of Shell6pElement
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(Shell6pElement);
+
+    /// Size types
+    typedef std::size_t IndexType;
+
+    // GometryType
```

Comment:
These annotations (also the ones for the constructors) provide 0 additional guidance. Either remove them or write proper documentation.